cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/controller/BoardsController.kt

```diff
@@ -0,0 +1,65 @@
+package com.cognifide.cogboard.config.controller
+
+import com.cognifide.cogboard.CogboardConstants
+import com.cognifide.cogboard.config.EndpointLoader
+import com.cognifide.cogboard.storage.Storage
+import com.cognifide.cogboard.storage.docker.VolumeStorage
+import com.cognifide.cogboard.widget.Widget
+import com.cognifide.cogboard.widget.WidgetIndex
+import io.vertx.core.AbstractVerticle
+import io.vertx.core.json.JsonObject
+import io.vertx.core.logging.Logger
+import io.vertx.core.logging.LoggerFactory
+
+internal open class BoardsController : AbstractVerticle() {
+
+    private lateinit var storage: Storage
+    internal val widgets = mutableMapOf<String, Widget>()
+
+    override fun start() {
+        storage = VolumeStorage(vertx)
+        listenOnBoardsConfigSave()
+        loadBoardsConfig()
+    }
+
+    private fun listenOnBoardsConfigSave() = vertx
+            .eventBus()
+            .consumer<JsonObject>(CogboardConstants.EVENT_SAVE_BOARDS_CONFIG)
+            .handler { storage.saveBoardsConfig(it.body()) }
+
+    private fun loadBoardsConfig() = storage
+            .loadBoardsConfig()
+            .getJsonObject(CogboardConstants.PROP_WIDGETS)
+            .getJsonObject(CogboardConstants.PROP_WIDGETS_BY_ID)
+            .forEach {
+                createOrUpdate(JsonObject(it.value.toString()))
+            }
+
+    fun createOrUpdate(config: JsonObject) {
+        var newConfig = config
+        val id = config.getString(CogboardConstants.PROP_ID)
+
+        if (id != null) {
+            widgets[id]?.let {
+                it.stop()
+                newConfig = it.config().mergeIn(config, true)
+            }
+            newConfig.attachEndpoint()
```

Comment:
I added extension function here as well for better comprehension and to stick with Kotlin best practices 😉